syncer/syncer.go

```diff
@@ -241,7 +242,8 @@ func NewSyncer(cfg *config.SubTaskConfig, etcdClient *clientv3.Client) *Syncer {
 	syncer.binlogSizeCount.Store(0)
 	syncer.lastCount.Store(0)
 	syncer.count.Store(0)
-	syncer.c = newCausality()
+	syncer.causality = newCausality(cfg.DisableCausality, cfg.QueueSize, cfg.Name, cfg.SourceID, &logger)
+	syncer.dmlWorker = newDMLWorker(cfg.Batch, cfg.WorkerCount, cfg.QueueSize, &logger, cfg.Name, cfg.SourceID, cfg.WorkerName, syncer.successFunc, syncer.fatalFunc, syncer.updateReplicationJobTS, syncer.addCount)
```

Comment:
i think don't need to make dmlworker as syncer's member because it only used with `causalityCh`

Reply:
It also use the logger created in this function. I think it's ok, just like shardingGroup, errOperatorHolder